src/main/java/learnositysdk/request/Init.java

```diff
@@ -376,7 +400,7 @@ private void setRequestPacket(Object requestPacket) throws Exception
         JSONObject convertedPacket = validateRequestPacket(requestPacket);
 
         // Prevent updating request packet in situations where we must not do so
-        if (!this.canAlterRequestData) {
+        if (this.preserveRequest) {
```

Comment:
Can we log some sort of warning about not using `DataAPI` here? @klauste what would be the best way to show a warning?

Reply:
I'm not sure a warning is necessary for now. I think we will need to establish what direction we want to go in the future (i.e. during our upcoming SDK audit) first to make sure we actually have a suitable alternative beforehand.